### DISCUSSION TITLE
Hidden Form Fields in Table Header Actions on Submit

### DISCUSSION THREAD

**Discussion by xdiegom** [filamentphp/filament#10678] [General]
Hi community!

I am facing a situation where I am using a Custom Header Action that is used in a Relation Manager. 

The situation is that the form that I am defining contains a hidden field that I want to use when the form is submitted and because is hidden, I don't want the user to see it. 

Basically, the hidden field is a TextInput that contains an ID of the parent (ownerRecord) of the RelationManager and once the form is submitted, only the visible fields are sent into the action hook.

This is the snippet of my custom Header Action:

```php
class CustomHeaderAction
{

    public static function getDefaultName(): ?string
    {
        return 'custom_header_action';
    }

    protected function setUp(): void
    {
        parent::setUp();
        
        $this->form([
            Forms\Components\TextInput::make('owner_record_id')
                ->hidden(),
            Grid::make()->schema([
                Forms\Components\TextInput::make('owner_record_field')
                    ->disabled(),
                Forms\Components\Select::make('my_model_id')
                    ->options(
                        function (Get $get) {
                            return MyModel::where('owner_record_id', '!=', $get('owner_record_id'))
                                ->get()
                                ->pluck('name', 'id')
                                ->toArray();
                        }
                    ),
                Forms\Components\TextInput::make('another_field')
                    ->required()
                    ->numeric()
                    ->minValue(1)
            ])->columns(3)

        ]);
        
       $this->action(function (array $arguments, array $data): void {
            dd($data);
            return;
        });
    }

}
```

RelationManager table function:

```php
    public function table(Table $table): Table
    {
        return $table
            ->columns([
                // Table columns
            ])
            ->modifyQueryUsing(fn ($query) => $query->orderBy('created_at', 'desc'))
            ->filters([
                //
            ])
            ->headerActions([
                CustomHeaderAction::make()
                    ->fillForm([
                        'owner_record_id' => $this->ownerRecord->id,
                        'owner_record_field' => $this->ownerRecord->name
                    ]),
                Tables\Actions\CreateAction::make()
            ])
            ->actions([

            ])
            ->bulkActions([
                //
            ]);
    }
```

The `dd()` that is inside the `action()` hook contains the form data that is sent after submit, right now the ones that are sent are:
- my_model_id
- another_field

which is good, because I need them. The only one that is missing and that I want is `owner_record_id` which is the hidden field. 

**Key takeway**

Inside the Select `options()` field method, the `$get` variable is fetching the `owner_id` record to exclude the owner record id from the `MyModel` class, and it works as expected. 

I have tried to pass arguments to the action via the `hedaersActions()` method inside the table function in the Relation manager but I did not have success.

I wonder if you could help me know if there is a way to handle hidden fields to be considered in form submition!

Thanks!

**Comment by leandrocfe** [comment]
Did you try the Hidden Field?
https://filamentphp.com/docs/3.x/forms/fields/hidden

**Reply by xdiegom** [reply]
Hi @leandrocfe , 

Yes, but did not work. If you notice, the snippet code I shared has `->hidden()` method in the form schema definition:
```php
Forms\Components\TextInput::make('owner_record_id')
                ->hidden(),
```

The problem that I am facing is that the hidden field is not sent after form is submitted.

**Reply by leandrocfe** [reply]
I mean `Hidden::make('owner_record_id')`

**Reply by xdiegom** [reply]
I'll give it a try!

**Reply by abderraxim** [reply]
> I mean `Hidden::make('owner_record_id')`

Thanks Mate
